org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -367,6 +375,25 @@ private void applyTrustManagers() throws SSLException {
             if (SSL.ConfigJSSDefaultCertAuthCallback(ssl_fd) == SSL.SECFailure) {
                 throw new SSLException("Unable to configure JSSNativeTrustManager on this JSSengine: " + errorText(PR.GetError()));
             }
+            return;
+        }
+
+        if (as_server) {
+            // We need to manually invoke the async cert auth handler. However,
+            // SSLFDProxy makes this easy for us: our CertAuthHandler derives
+            // from Runnable, so we can reuse it here as well.
```

Comment:
Mention that we need to create it ahead of time though, rather than via `checkNeedCertValidation`